client-tags/message-status.md

```diff
@@ -0,0 +1,83 @@
+---
+title: IRCv3 `message-status` client tags
+layout: spec
+work-in-progress: true
+copyrights:
+  -
+    name: "Evan Magaliff"
+    period: "2017"
+    email: "evan@muffinmedic.net"
+---
+## Purpose
+This specification defines several `message-status` client tags, which allows clients to send and receive various information about the status of messages before and after they are sent.
+
+## Description
+The specification includes three client tags for message tracking: `typing`, `delivered`, and `read`. The `typing` tag indicates that a user is currently typing a message to a specific target. `delivered` and `read` provide for delivery and read receipts, respectively, and enable users to know when and by whom a message has been delivered to or read.
+
+For the purpose of this specification, the terms "delivery receipt" and "read receipt" are synonymous with the use of the `delievered` and `read` client tags, respectively.
+
+## Format
+All tags are sent as a [`TAGMSG`][tags] with an appropriate [`server-time`][time]. `delivered` and `read` must also include the [`labeled-response` tag][labels] of the original message. `typing` can be either `true` or `false`; `delivered` and `read` contain no values.
+
```

Comment:
Worth mentioning that clients should consider all previous messages up to the indicated msgid read/delivered when receiving such a message. And also not to ever go "back in time"

For example:

```
C1: @msgid=123 PRIVMSG #chan :hi there
C1: @msgid=456 PRIVMSG #chan :hi there again
C2: @delivered=456;read=456 TAGMSG #chan
C2: @delivered=123;read=123 TAGMSG #chan
```

C2 has received/read up to message `456`, the second TAGMSG for `123` can be ignored as superfluous.

This doesn't prevent clients from maintaining their own concept of what has been read, i.e. some clients might want to let users artificially move their read marker back in time as a reference, but that wouldn't be meaningfully propagated to anyone else.